cpp/vm/evmzero/interpreter.cc

```diff
@@ -1487,6 +1489,24 @@ struct CreateImpl {
     if (gas -= mem_cost; gas < 0) [[unlikely]]
       return {.dynamic_gas_costs = initial_gas - gas};
 
+    if (ctx.revision >= EVMC_SHANGHAI) {
+      const int64_t kMaxCodeSize = 24576;
+      const int64_t kMaxInitCodeSize = 2 * kMaxCodeSize;
+      const int64_t kInitCodeWordCost = 2;
+
+      if (init_code_size > kMaxInitCodeSize) {
+        return {
+            .state = RunState::kErrorInitCodeSizeExceeded,
+            .dynamic_gas_costs = initial_gas - gas,
```

Comment:
should the cost still be updated on a failed case ?

Reply:
This is also performed in all the other operations, all gas has to be consumed in a failed case.